README.md

```diff
@@ -22,12 +22,12 @@ yarn add toucan-sdk
 
 # Quickstart
 
-Instantiate the Toucan client to interact with our infrastructure.
+Instantiate the ToucanClient to interact with our infrastructure.
 
 ```typescript
-import ToucanClient from "toucan-sdk";
+import { ToucanClient } from "toucan-sdk";
 
-const toucan = new ToucanClient("polygon", provider, signer);
```

Comment:
Why is the signer and provider omitted here?

Reply:
I thought the point of having the separate clients was that a user that does only need to use the Subgraph can use `SubgraphInteractions` directly without having to instantiate a `ToucanClient`?

Reply:
Personally I still see value in providing the signer during initialization and not having to pass in functions. In general we need to think about how often will a developer use the SDK in their app only for functions that don't need a signer vs how often they will need to provide one and we should pick the choice that makes happy most of the developers.

As a compromise to supporting both options we could go a step further and have a separate `ToucanReadOnlyClient` that works without a signer but may be an overkill for this PR and simply making the `signer` optional in the constructor would satisfy both cases for now.

Reply:
From my experience with working with other SDKs, I have found it best not to have to deal with a signer either at all ([Lyra SDK](https://github.com/lyra-finance/lyra-js)) or as little as possible ([Primitive SDK](https://github.com/primitivefinance/rmm-sdk)). I'd imagine having to have a signer handy every time I need to use a function would be annoying since most of the use in apps I have worked is usually a mix of writes and reads. I agree that setting a TCO2 in our case as part of the client is problematic but don't think this is the case with a signer and I have never had the need to use multiple signers within the same class instance in any of the apps I have built.

Reply:
Btw fine with a setter, too. I think my main issue is with forcing it as an argument in all write functions when most developers will most likely just need to instantiate it once and not want to deal with it again.

Reply:
love the discussion and insights! the readonly client is an interesting approach we could definitely investigate down the line if we get feedback indicating people would use/need it. 

how does signing work with the lyra sdk @kargakis?